lib/foreman/renderer/scope/macros/base.rb

```diff
@@ -334,20 +334,33 @@ def gem_version_compare(first, second)
             Gem::Version.new(first.to_s) <=> Gem::Version.new(second.to_s)
           end
 
-          apipie :method, "Returns content of 'SSL CA file' configured in Settings > Authentication" do
+          apipie :method, "Returns a combination of 'Server CA file' and 'SSL CA file' configured in Settings > Authentication" do
             example "SSL_CA_CERT=$(mktemp)
                      cat > $SSL_CA_CERT <<CA_CONTENT
                      <%= foreman_server_ca_cert %>
                      CA_CONTENT
-                     curl --cacert $SSL_CA_CERT https://smart-proxy.example.com:8443"
-          end
-          def foreman_server_ca_cert
-            if File.exist?(Setting[:ssl_ca_file])
-              File.read(Setting[:ssl_ca_file])
-            else
-              msg = N_("SSL CA file not found, check the 'SSL CA file' in Settings > Authentication")
-              raise Foreman::Exception.new(msg)
+                     curl --cacert $SSL_CA_CERT https://foreman.example.com:8443"
+          end
+          def foreman_server_ca_cert(server_ca_file_enabled: true, ssl_ca_file_enabled: true)
+            setting_names = [].tap do |array|
+              array << :server_ca_file if server_ca_file_enabled
+              array << :ssl_ca_file if ssl_ca_file_enabled
+            end
+
+            setting_values = setting_names.map { |setting_name| Setting[setting_name] }.compact
+
+            files_content = setting_values.map do |setting_value|
+              File.read(setting_value)
+            rescue Errno::ENOENT, Errno::EACCES
+              nil
             end
+
+            result = files_content.compact.join("\n")
+
+            return result if result.present?
+
+            msg = N_("SSL CA file not found, check the 'Server CA file' and 'SSL CA file' in Settings > Authentication")
+            raise Foreman::Exception.new(msg)
```

Comment:
Nit: I'm not a fan of the flow this way. To me it reads as that the default flow is an exception while returning the result is the expected flow. Not sure how others read it.